superset-frontend/src/dashboard/components/Header.jsx

```diff
@@ -249,14 +272,21 @@ class Header extends React.PureComponent {
       colorNamespace,
       colorScheme,
       dashboardInfo,
-      refreshFrequency,
+      refreshFrequency: currentRefreshFrequency,
+      isPersistentRefreshFrequency,
     } = this.props;
 
     const scale = CategoricalColorNamespace.getScale(
       colorScheme,
       colorNamespace,
     );
     const labelColors = colorScheme ? scale.getColorMap() : {};
+    // check refresh frequency is for current session or persist
+    const refreshFrequency = getPersistentRefreshFrequency({
+      currentRefreshFrequency,
+      isPersistent: isPersistentRefreshFrequency,
+      dashboardInfo,
+    });
```

Comment:
I don't think we need a util function for this. It might be more straightforward to simply do

```js
const refreshFrequency = isPersistentRefreshFrequency
  ? currentRefreshFrequency
  : dashboardInto.metadata.refresh_frequency;
```

Reply:
this logic used in 2 files, save button and save modal. I feel isolate out this small piece is better than copy and paste.

Reply:
But you'd need to copy & paste the function calls nonetheless?

Reply:
ok. i agree to make this logic inline (removed the util function).